app/src/main/java/com/example/project/MapsFragment.kt

```diff
@@ -88,32 +118,138 @@ class MapsFragment : Fragment(), OnMapReadyCallback {
         getLocationPermission()
 
         // if location is not turned on direct user to settings
-        if(checkIfLocationIsOn() && locationPermissionGranted){
-            activity?.supportFragmentManager?.let { LocationDialogFragment().show(it, "TAG") }
-        }
+        directUserToLocationSettings()
 
         // Turn on locations settings on map
         updateLocationUI()
 
         // Get the current location of the device and set the position of the map.
         getDeviceLocation()
+
+        handleRestaurantSearch()
+
+    }
+
+    private fun handleRestaurantSearch(){
+        searchBar.addTextChangeListener(object: TextWatcher{
+            val token = AutocompleteSessionToken.newInstance()
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                //show a list of suggestions or autocomplete
+                val predictionsRequest = FindAutocompletePredictionsRequest.builder()
+                    .setCountry("fi")
+                    .setOrigin(lastKnownLocation?.latitude?.let { lastKnownLocation?.longitude?.let { it1 ->
+                        LatLng(it,
+                            it1
+                        )
+                    } })
+                    .setTypeFilter(TypeFilter.ESTABLISHMENT)
+                    .setSessionToken(token)
+                    .setQuery(p0.toString())
+                    .build()
+                    placesClient.findAutocompletePredictions(predictionsRequest).addOnCompleteListener {task ->
+                        if (task.isSuccessful){
+                            val predictionResponse = task.result
+                            if (predictionResponse != null){
+                                predictionsList = predictionResponse.autocompletePredictions
+                                val autoCompleteResults = ArrayList<String>()
+                                predictionsList.forEach {
+                                    autoCompleteResults.add(it.getFullText(null).toString())
+                                }
+                                searchBar.updateLastSuggestions(autoCompleteResults)
+                            }
+                        }else{
+                            Log.d(Constants.TAG, "fetching predictions failed")
+                        }
+                    }
+            }
+
+            override fun afterTextChanged(p0: Editable?) {
+            }
+
+        })
+
+
+        searchBar.setSuggestionsClickListener(object: SuggestionsAdapter.OnItemViewClickListener{
```

Comment:
Besides this, there should be an implementation of handling the following cases
- Users just typing in the location and pressing enter without selecting the suggestions.
- After the users cancel the search result by clicking on the X icon in the search bar, the marker corresponding to that location should be cleared as well

If you plan to implement this in further tasks (cards), please create them (in case you forget) and ignore this comment